File: Javascript/Algorithms/1614.js
let maxDepth = s => {
    let depth = 0, max = 0;
    for (let i of s){
        if (i == '('){
            depth++;
            if (depth > max) {
                max = depth;
            }
        }
        if (i == ')'){
            depth--;
        }
    }
    return max;
};

File: Rust/Algorithms/1614.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let (mut max, mut depth) = (0, 0);
        for c in s.chars() {
            match c {
                '(' => {
                        depth += 1;
                        if depth > max {
                            max = depth;
                        }
                },
                ')' => {
                    depth -= 1;    
                },
                _ => (),
            }
        }
        max
    }
}
